javaparser-core/src/main/java/com/github/javaparser/ast/CompilationUnit.java

```diff
@@ -465,6 +469,26 @@ public AnnotationDeclaration addAnnotationDeclaration(String name, Modifier... m
         return getTypes().stream().filter(type -> type.getNameAsString().equals(enumName) && type instanceof EnumDeclaration).findFirst().map(t -> (EnumDeclaration) t);
     }
 
+    /**
+     * @return the name that the primary type in this file should have, according to the filename in {@link Storage#getFileName()}.
+     * Empty if no file information is present (when this compilation unit wasn't parsed from a file.)
+     */
+    public Optional<String> getPrimaryTypeName() {
```

Comment:
Would be something like this?
``` java
CompilationUnit compU = JavaParser.parse(file);
ClassOrInterfaceDeclaration = compU.getClassByName(getPrimaryTypeName.get());
```

Reply:
That method has been added too: `getPrimaryType`